src/superface-test.ts

```diff
@@ -0,0 +1,393 @@
+} from '@superfaceai/one-sdk';
+import { createHash } from 'crypto';
+import {
+  activate as activateNock,
+  define,
+  disableNetConnect,
+  enableNetConnect,
+  isActive as isNockActive,
+  recorder,
+  restore as restoreRecordings,
+} from 'nock';
+import { join as joinPath } from 'path';
+
+import {
+  ProcessingFunction,
+  RecordingDefinitions,
+  RecordingProcessOptions,
+} from '.';
+import {
+  ComponentUndefinedError,
+  FixturesPathUndefinedError,
+  MapUndefinedError,
+  RecordingPathUndefinedError,
+  RecordingsNotFoundError,
+  UnexpectedError,
+} from './common/errors';
+import {
+  getFixtureName,
+  matchWildCard,
+  removeTimestamp,
+} from './common/format';
+import { exists, readFileQuiet } from './common/io';
+import { writeRecordings } from './common/output-stream';
+import {
+  NockConfig,
+  SuperfaceTestConfigPayload,
+  SuperfaceTestRun,
+  TestingReturn,
+} from './superface-test.interfaces';
+import {
+  assertBoundProfileProvider,
+  assertsDefinitionsAreNotStrings,
+  assertsPreparedConfig,
+  getProfileId,
+  getProviderName,
+  getSuperJson,
+  isProfileProviderLocal,
+  replaceCredentials,
+} from './superface-test.utils';
+
+export class SuperfaceTest {
+  private sfConfig: SuperfaceTestConfigPayload;
+  private boundProfileProvider?: BoundProfileProvider;
+  private nockConfig?: NockConfig;
+  private fixturesPath?: string;
+  private recordingPath?: string;
+
+  constructor(sfConfig?: SuperfaceTestConfigPayload, nockConfig?: NockConfig) {
+    this.sfConfig = sfConfig ?? {};
+    this.nockConfig = nockConfig;
+
+    this.setupFixturesPath();
+  }
+
+  /**
+   * Sets up path to all fixtures.
+   */
+  private setupFixturesPath(): void {
+    const { path } = this.nockConfig ?? {};
+
+    if (this.fixturesPath === undefined) {
+      this.fixturesPath = path ?? joinPath(process.cwd(), 'nock');
+    }
+  }
+
+  /**
+   * Sets up path to recording, depends on current Superface configuration.
+   */
+  private setupRecordingPath(fixtureName: string, inputHash: string) {
+    if (!this.fixturesPath) {
+      throw new FixturesPathUndefinedError();
+    }
+
+    this.recordingPath = joinPath(
+      this.fixturesPath,
+      fixtureName,
+      `${this.nockConfig?.fixture ?? 'recording'}-${inputHash}.json`
+    );
+  }
+
+  /**
+   * Tests current configuration whether all necessary components
+   * are defined and ready to use and tries to perform entered usecase.
+   */
+  async run(
+    testCase: SuperfaceTestRun,
+    options?: RecordingProcessOptions
+  ): Promise<TestingReturn> {
+    this.prepareSuperfaceConfig(testCase);
+    await this.setupSuperfaceConfig();
+
+    assertsPreparedConfig(this.sfConfig);
+    if (!(await this.checkForMapInSuperJson())) {
+      throw new MapUndefinedError(
+        getProfileId(this.sfConfig.profile),
+        getProviderName(this.sfConfig.provider)
+      );
+    }
+
+    this.boundProfileProvider =
+      await this.sfConfig.client.cacheBoundProfileProvider(
```

Comment:
`cacheBoundProfileProvider` is deprecated

Reply:
this is okay, we talked about this and i'll fix it when i remove cacheBoundProfileProvider

Reply:
👍 ok thanks